test/utils/int/providerRemote.ts

```diff
@@ -20,3 +26,44 @@ export async function clearStoredNotifications(windowIdentity: Identity): Promis
         });
     }, windowIdentity);
 }
+
+/**
+ * Closes all of the service providers child windows and then restarts the main window. Promise resolves once the service is fully initialized.
+ * This will not reopen any child windows unless the provider automatically does so.
+ */
+export async function restartProvider(): Promise<void> {
+    const providerApp = fin.Application.wrapSync(serviceIdentity);
+    await providerApp.getChildWindows().then(children => children.forEach(win => win.close()));
+    await providerApp.restart();
+
+    await providerReady();
+}
+
+/**
+ * Checks that the provider is ready and has been initialized.
+ */
+export async function providerReady(): Promise<void> {
+    const TIMEOUT = 5000;
+    const RETRY_TIMEOUT = 250;
+
+    await new Promise((resolve, reject) => {
+        let counter: number = 0;
+
+        setInterval(async () => {
+            // Keep retrying until injector is initialized
+            await ofBrowser.executeOnWindow(serviceIdentity, async function() {
+                await this.injector.initialized;
```

Comment:
This will start hitting the provider as soon as the app is restarted, so its a real grab as to what errors to expect.  Sometimes `this.injector.initialized` might not be a thing, etc. We don't really care about the errors, only that the provider is up is initialized.  If there are other errors, then there are bigger problems. If the `RETRY_TIMEOUT` elapses then I we would reject and expect the test to fail. Ideally 5s should be plenty of time and again if this fails then there are bigger problems.